src/Bernard/Silex/BernardServiceProvider.php

```diff
@@ -12,165 +12,160 @@
+        $container['bernard.consumer_middleware'] = function ($container) {
             return new Middleware\MiddlewareBuilder;
-        });
+        };
 
-        $app['bernard.producer_middleware'] = $app->share(function ($app) {
+        $container['bernard.producer_middleware'] = function ($container) {
             return new Middleware\MiddlewareBuilder;
-        });
-
+        };
 
         // defaults.
         foreach (array('bernard.receivers', 'bernard.options') as $default) {
-            if (!isset($app[$default])) {
-                $app[$default] = array();
+            if (!isset($container[$default])) {
+                $container[$default] = array();
             }
         }
     }
 
     /**
-     * {@inheritDoc}
+     * @param Container $container
      */
-    public function boot(Application $app)
+    protected function registerSerializers(Container $container)
     {
-    }
-
-    /**
-     * @param Application $app
-     */
-    protected function registerSerializers(Application $app)
-    {
-        $app['bernard.serializer_simple'] = $app->share(function () {
+        $container['bernard.serializer_simple'] = function () {
             return new Serializer\SimpleSerializer;
-        });
+        };
 
-        $app['bernard.serializer_symfony'] = $app->share(function ($app) {
-            return new Serializer\SymfonySerializer($app['serializer']);
-        });
+        $container['bernard.serializer_symfony'] = function ($container) {
+            return new Serializer\SymfonySerializer($container['serializer']);
+        };
 
-        if (isset($app['serializer'])) {
-            $app['serializer.normalizers'] = $app->share($app->extend('serializer.normalizers', function ($normalizers) {
+        if (isset($container['serializer'])) {
+            $container['serializer.normalizers'] = $container->extend('serializer.normalizers', function ($normalizers) {
                 array_unshift($normalizers, new Symfony\EnvelopeNormalizer, new Symfony\DefaultMessageNormalizer);
 
                 return $normalizers;
-            }));
+            });
         }
 
-        $app['bernard.serializer_jms'] = $app->share(function ($app) {
-            return new Serializer\JMSSerializer($app['jms_serializer']);
-        });
+        $container['bernard.serializer_jms'] = function ($container) {
+            return new Serializer\JMSSerializer($container['jms_serializer']);
+        };
 
-        if (isset($app['jms_serializer'])) {
-            $app['jms_serializer.builder'] = $app->share($app->extend('jms_serializer.builder', function ($builder, $app) {
+        if (isset($container['jms_serializer'])) {
+            $container['jms_serializer.builder'] = $container->extend('jms_serializer.builder', function ($builder, $container) {
                 $builder->configureHandlers(function ($registry) {
                     $register->registerSubscribingHandler(new JMSSerializer\EnvelopeHandler);
```

Comment:
@andytruong hey, `$register` here is not defined. There must be something wrong here :)